src/kubelet/client/connector.go

```diff
@@ -228,6 +230,18 @@ func checkConnection(conn connParams) error {
 	return nil
 }
 
+func (dp *defaultConnector) pesterClientWithTimeout(timeout time.Duration) *pester.Client {
```

Comment:
I have some concerns on this method being part of the `defaultConnector`. I think the connector should do as few things as possible, namely returning `connParams` on `Connect()`.

Since pester supports wrapping an existing client (i.e. `pester.NewExtendedClient()`), do you think it might be better to use that?

Something like

```go
func (cp connParams) WithPester() connParams {
	cp.client = pester.NewExtendedClient(cp.client) // Ugly cast ellided for brevity
	return cp
}
```

We can also put that functionality outside of `connParams` if you think it's cleaner, I'm not fully in for baking it in.

Reply:
you are right, changing completely the approach